### PR TITLE
feat: Added platform expense services

### PR DESCRIPTION
**Pull request by OmkarJ13**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=4N6Rf6t)
### <samp>🤖 Generated by Copilot at f7747d0</samp>

This pull request adds services and tests for the approver and spender roles in the platform API. It also adds a mock data file for the expense model. The services are wrappers for making HTTP requests to the API endpoints, and the tests verify their functionality. The mock data file is used for testing purposes. The files affected are `src/app/core/mock-data/platform/v1/expense.data.ts`, `src/app/core/services/platform/v1/approver/approver.service.spec.ts`, `src/app/core/services/platform/v1/approver/approver.service.ts`, `src/app/core/services/platform/v1/approver/expenses.service.spec.ts`, `src/app/core/services/platform/v1/approver/expenses.service.ts`, `src/app/core/services/platform/v1/spender/expenses.service.spec.ts`, `src/app/core/services/platform/v1/spender/expenses.service.ts`, `src/app/core/services/platform/v1/spender/spender.service.spec.ts`, and `src/app/core/services/platform/v1/spender/spender.service.ts`.

### <samp>🤖 Generated by Copilot at f7747d0</samp>

> _`Approver`, `spender`_
> _Services and tests for expenses_
> _Mock data in spring_

### <samp>🤖 Generated by Copilot at f7747d0</samp>

*  Add mock data file for expense model for testing purposes ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bR1-R354))
*  Add and implement approver service for making HTTP requests to platform API endpoints related to approver role ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-5386a205bfbe6f004229cd9e5bc001ebb1aa58658c3f1ef7d14eaabd64f1fff0R1-R27))
   * Add unit tests for get and post methods of approver service ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-7cf384dc86d7c95af6daaf7a77d914c7148f21f0b2f460fb61d0f051a1aa798aR1-R94))
   * Inject approver service as a dependency in expenses service under approver role ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-f4e4d05a0d75d9da23b0b202369453f3790a074d7e7e5782030db3045cf1b377R1-R20))
*  Add and implement expenses service under approver role for fetching and manipulating expenses for approver ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-f4e4d05a0d75d9da23b0b202369453f3790a074d7e7e5782030db3045cf1b377R1-R20))
   * Add unit test for getById method of expenses service under approver role ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-fb8d034982aebe6d33809c5f68618313420da5719c0a690fe3f0ec88e3dc69bfR1-R38))
*  Add and implement spender service for making HTTP requests to platform API endpoints related to spender role ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-5ab49009500c128dbd986146e9217404593ae74bfbed382fd24b7f298a29bcc7R1-R27))
   * Add unit tests for get and post methods of spender service ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-464ee5d178d735c8b147af7a12e7ae29e227971eee46fa453185c5d9a470717fR1-R94))
   * Inject spender service as a dependency in expenses service under spender role ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-43a6608834bfbe4963726f4de34dd6082b69581858020acfa23be1e61037d9f5R1-R20))
*  Add and implement expenses service under spender role for fetching and manipulating expenses for spender ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-43a6608834bfbe4963726f4de34dd6082b69581858020acfa23be1e61037d9f5R1-R20))
   * Add unit test for getById method of expenses service under spender role ([link](https://github.com/fylein/fyle-mobile-app/pull/2530/files?diff=unified&w=0#diff-7e1ed82416d113f0ad13f11d3238de7424dc72ac6b24ffe030dd7a66b2b139bfR1-R38))

## Clickup
https://app.clickup.com

### CODE REVIEW

File: src/app/core/services/router-api.service.ts [hunk 1]
```diff
@@ -12,11 +13,11 @@ export class RouterApiService {
     this.ROUTER_API_ENDPOINT = environment.ROUTER_API_ENDPOINT;
   }
 
-  post(url: string, data: any) {
-    return this.httpClient.post<any>(this.ROUTER_API_ENDPOINT + '/routerapi' + url, data);
+  post<T>(url: string, data = {}): Observable<T> {
```
Comment (OmkarJ13) [inline src/app/core/services/router-api.service.ts:16]: Linting fixes

File: src/app/core/services/router-auth.service.ts [hunk 2]
```diff
@@ -64,32 +70,34 @@ export class RouterAuthService {
     this.approverPlatformApiService.setRoot(domain);
     this.spenderPlatformV1ApiService.setRoot(domain);
     this.expenseAggregationService.setRoot(domain);
+    this.spenderService.setRoot(domain);
```
Comment (OmkarJ13) [inline src/app/core/services/router-auth.service.ts:73]: All the changes in this file are linting ones, only this is the actual change

File: src/app/core/models/app-supported-details.model.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export interface AppSupportedDetails {
```
Comment (OmkarJ13) [inline src/app/core/models/app-supported-details.model.ts:1]: Linting fixes

File: src/app/core/services/app-version.service.ts [hunk 4]
```diff
@@ -71,29 +72,31 @@ export class AppVersionService {
       .subscribe(noop); // because this needs to happen in the background
   }
 
-  getUserAppVersionDetails(deviceInfo: ExtendedDeviceInfo) {
+  getUserAppVersionDetails(deviceInfo: ExtendedDeviceInfo): Observable<{
```
Comment (OmkarJ13) [inline src/app/core/services/app-version.service.ts:75]: All linting fixes here

File: src/app/core/services/invitation-requests.service.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 
 @Injectable({
   providedIn: 'root',
 })
 export class InvitationRequestsService {
   constructor(private routerApiService: RouterApiService) {}
 
-  upsertRouter(email: string) {
-    return this.routerApiService.post('/invitation_requests/invite', { email });
+  upsertRouter(email: string): Observable<null> {
```
Comment (Julias0) [inline src/app/core/services/invitation-requests.service.ts:11]: Observable<null> -> Observable<void>

File: src/app/core/services/platform/v1/approver/expenses.service.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+  getById(id: string): Observable<Expense> {
+    const data = {
+      params: {
+        id: `eq.${id}`,
+      },
+    };
+
+    return this.approverService.get<PlatformApiResponse<Expense>>('/expenses', data).pipe(map((res) => res.data[0]));
+  }
```
Comment (SowmyaFyle) [inline src/app/core/services/platform/v1/approver/expenses.service.ts:21]: convert the date strings to date objects before returning the response

File: src/app/core/services/app-version.service.ts [hunk 7]
```diff
@@ -71,29 +72,31 @@ export class AppVersionService {
...
       lastLoggedInVersion: this.loginInfoService.getLastLoggedInVersion(),
       eou: from(this.authService.getEou()),
     }).pipe(
       filter(
-        (res: {
-          appSupportDetails: { supported: boolean; message?: string };
-          lastLoggedInVersion: string;
-          eou: ExtendedOrgUser;
-        }) => !res.appSupportDetails.supported && environment.production
+        (res: { appSupportDetails: AppSupportedDetails; lastLoggedInVersion: string; eou: ExtendedOrgUser }) =>
```
Comment (hlkavya0213) [inline src/app/core/services/app-version.service.ts:87]: better name for res

File: src/app/core/mock-data/platform/v1/expense.data.ts [hunk 8]
```diff
@@ -0,0 +1,354 @@
+import { ExpenseState } from 'src/app/core/models/expense-state.enum';
+import { ApprovalState } from 'src/app/core/models/platform/report-approvals.model';
+import { AccountType } from 'src/app/core/models/platform/v1/account.model';
+import { Expense } from 'src/app/core/models/platform/v1/expense.model';
+
+export const expenseData1: Expense = {
```
Comment (hlkavya0213) [inline src/app/core/mock-data/platform/v1/expense.data.ts:6]: why name it expenseData1? let's just use `expenseData` ?

File: src/app/core/services/platform/v1/approver/approver.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+  it('should make GET request with params', (done) => {
+    httpClient.get.and.returnValue(of(apiResponse));
+    dateService.fixDates.and.returnValue(apiResponse);
+
+    approverService
+      .get('/expense_fields', {
+        params: requestObj,
+      })
+      .subscribe((res) => {
+        expect(res).toEqual(apiResponse);
```
Comment (hlkavya0213) [inline src/app/core/services/platform/v1/approver/approver.service.spec.ts:75]: better naming pls, fix it everywhere

File: src/app/core/services/platform/v1/approver/expenses.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('getById(): should return expense with the given id', (done) => {
+    approverService.get.and.returnValue(of({ data: [expenseData1] }));
+    const expenseId = 'txOJVaaPxo9O';
+
+    service.getById(expenseId).subscribe((res) => {
+      expect(res).toBeTruthy();
```
Comment (hlkavya0213) [inline src/app/core/services/platform/v1/approver/expenses.service.spec.ts:29]: let's name variable properly

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import { PlatformApiResponse } from 'src/app/core/models/platform/platform-api-response.model';
+import { Expense } from 'src/app/core/models/platform/v1/expense.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExpensesService {
+  constructor(private spenderService: SpenderService) {}
+
+  getById(id: string): Observable<Expense> {
```
Comment (hlkavya0213) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:13]: getExpenseById?

File: src/app/core/services/platform/v1/approver/expenses.service.ts (hunk 6 above)
Comment (OmkarJ13) [reply src/app/core/services/platform/v1/approver/expenses.service.ts:23]: Doing this in the top-level, basically in spenderService and approverService

**Review by hlkavya0213 — APPROVED** [review]

**Review by SowmyaFyle — APPROVED** [review]

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**